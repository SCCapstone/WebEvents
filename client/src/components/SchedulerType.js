import React from "react";
//import "./CSS/Options.css";
import 'bootstrap/dist/css/bootstrap.min.css';
//import Button from 'react-bootstrap/Button';
import Dropdown from 'react-bootstrap/Dropdown';
import DropdownButton from 'react-bootstrap/DropdownButton';

import "../CSS/webevents-main.css";

class SchedulerType extends React.Component {
    constructor(props){
        super (props);
        this.state = {
            type: "",     
        };
        //this.type = this.setType.bind(this);
        //change
        
    } 

    render() {

        //options dropdown-button
        return(
            <div className="schedulerTypeCont">
                <div className="scheduleTypeBar">
                    {['right'].map(direction => (
                    <DropdownButton
                        drop={direction}
                        variant="secondary"
                        title={"Scheduler Type"}
                        id={`dropdown-button-drop-${direction}`}
                        key={direction}
                    >
                        <Dropdown.Item onClick = {() => this.props.onClick("seminar")}>Seminar Scheduler </Dropdown.Item>
                        <Dropdown.Item onClick = {() => this.props.onClick("field")}>Field Scheduler</Dropdown.Item>
                        <Dropdown.Item onClick = {() => this.props.onClick("work")}>Work Scheduler</Dropdown.Item>
                        {console.log(this.state.type)}
                        </DropdownButton>
                    ))} 
                    <h5>Selected Scheduler: {this.props.SchedulerType}</h5>
                </div>
            </div>
        );
             
       // console.log(this.props.type);
    };
    //console.log(this.state.type);
}
//console.log(this.state.type);

/*<DropdownButton id="dropdown-basic-button" title="Scheduler Type">
                    <Dropdown.Item href="#/Scheduler 1">Seminar Type</Dropdown.Item>
                    <Dropdown.Item href="#/Scheduler 2">Field Scheduler</Dropdown.Item>
                    <Dropdown.Item href="#/Scheduler 3">Something else</Dropdown.Item>
                </DropdownButton>*/



/*class GroupNumber extends React.Component{


    render() {
        return;
    };
}

<ButtonToolbar>
                {['right'].map(direction => (
                <DropdownButton
                    drop={direction}
                    variant="secondary"
                    title={` Drop ${direction} `}
                    id={`dropdown-button-drop-${direction}`}
                    key={direction}
                >

                    
                    <Dropdown.Item onClick = {this.state.type = "seminar"}>Seminar Type </Dropdown.Item>
                    
                    <Dropdown.Item href="#/Scheduler 2">Field Scheduler</Dropdown.Item>
                    <Dropdown.Item href="#/Scheduler 3">Something else</Dropdown.Item>
                    </DropdownButton>
                ))} 
                
            </ButtonToolbar>
*/
export default SchedulerType;