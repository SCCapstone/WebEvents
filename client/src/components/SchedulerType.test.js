import React from "react";
import SchedulerType from "./SchedulerType";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";

//makes sure that the app renders without crashing
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SchedulerType />, div);
  ReactDOM.unmountComponentAtNode(div);
});
//makes sure that the sheet component matches the snapshot
describe("sheets component", () => {
  test("Matches the snapshot", () => {
    const type = create(<SchedulerType />);
    expect(type.toJSON()).toMatchSnapshot();
  });
});
