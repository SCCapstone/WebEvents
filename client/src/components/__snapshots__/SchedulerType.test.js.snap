// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sheets component Matches the snapshot 1`] = `
<div
  className="schedulerTypeCont"
>
  <div
    className="scheduleTypeBar"
  >
    <div
      className="dropright"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-secondary"
        disabled={false}
        id="dropdown-button-drop-right"
        onClick={[Function]}
        type="button"
      >
         Scheduler Type 
      </button>
    </div>
    <h5>
      Selected Schedule Type: 
    </h5>
  </div>
</div>
`;
